### PR TITLE
delete edit checkbox vuex statepersistance

### PR DESCRIPTION
**Pull request by DevkantExcel**
Vue router

### CODE REVIEW

File: vuestate/src/components/HelloWorld.vue [hunk 1]
```diff
@@ -12,71 +12,123 @@
...
 
 <script>
 //buefy css files
-import Vue from 'vue'
-import Buefy from 'buefy'
-import 'buefy/dist/buefy.css'
-
-Vue.use(Buefy)
+import Vue from "vue";
+import Buefy from "buefy";
```
Comment (reactexcel) [inline vuestate/src/components/HelloWorld.vue]: we don't need to use buefy anymore use this instead https://vuetifyjs.com/en/

File: vuestate/src/components/HelloWorld.vue [hunk 2]
```diff
@@ -12,71 +12,123 @@
...
 export default {
-  name: 'HelloWorld',
-  data(){
-    return{
-      todo: null
-    }
+  name: "HelloWorld",
+  data() {
+    return {
+      todo: null,
```
Comment (reactexcel) [inline vuestate/src/components/HelloWorld.vue]: we should not have any data if we are using vuex. this is fully wrong.

File: vuestate/src/components/HelloWorld.vue [hunk 3]
```diff
@@ -12,71 +12,123 @@
...
-    ...mapActions(['add','delete']),
-    addItem: function(){
-      if(this.todo !== null && this.todo !== '') {
-      this.add(this.todo)
-      this.todo = null
+    ...mapActions(["add", "delete", "edit"]),
+    addItem: function() {
+      if (this.todo !== null && this.todo !== "" && this.editIndex == null) {
+        this.add({
+          key00: this.todo,
```
Comment (reactexcel) [inline vuestate/src/components/HelloWorld.vue]: what is this key00, 01, etc. use proper names this doesnt make sense

File: vuestate/src/components/HelloWorld.vue [hunk 4]
```diff
@@ -12,71 +12,123 @@
...
+          key01: this.active,
+          key02: this.arrIndex
+        });
+        this.arrIndex += 1;
+        this.todo = null;
+      } else {
+        this.edit({
+          key1: this.editIndex,
```
Comment (reactexcel) [inline vuestate/src/components/HelloWorld.vue]: what is this again key1, key2??

File: vuestate/src/store/modules/store.js [hunk 5]
```diff
@@ -1,29 +1,69 @@
...
+        sort_desc: true,
     },
     mutations: {
-        addtodo: (state,item) =>{
-            // console.log(item)
-            state.todos.push(item)
-            console.log(state.todos)
+        addtodo: (state, item) => {
+            // console.log(item.key01)
+            state.todos.push({
```
Comment (reactexcel) [inline vuestate/src/store/modules/store.js]: this should be like state.todos.push(item)
this simple. pass proper json from component

File: vuestate/src/store/modules/store.js [hunk 6]
```diff
@@ -1,29 +1,69 @@
...
+                'sno': item.key02
+            })
+            // console.log(state.todos)
         },
         delItem: (state, index) => {
             state.todos.splice(index, 1)
+        },
+        editItem: (state, val) => {
+            // console.log(state.todos, val.key1, val.key2)
+            Vue.set(state.todos, val.key1, {
```
Comment (reactexcel) [inline vuestate/src/store/modules/store.js]: why using Vue.set whas the reason

File: vuestate/src/store/modules/store.js [hunk 7]
```diff
@@ -1,29 +1,69 @@
...
             commit('addtodo', payload)
         },
-        delete({commit}, payload) {
+        delete({
+            commit
+        }, payload) {
             commit('delItem', payload)
+        },
+        edit({
+            commit
```
Comment (reactexcel) [inline vuestate/src/store/modules/store.js]: this doesnt look formatted. all functions should be in single line

File: vuestate/src/store/modules/store.js (hunk 6 above)
Comment (DevkantExcel) [reply vuestate/src/store/modules/store.js]: to update array elements I used Vue.set

**Pull request merged**